charts/kube-state-metrics/templates/deployment.yaml

```diff
@@ -91,12 +91,17 @@ spec:
         - --telemetry-host={{ .Values.selfMonitor.telemetryHost }}
         {{- end }}
         - --telemetry-port={{ .Values.selfMonitor.telemetryPort | default 8081 }}
-        {{- if .Values.kubeconfig.enabled }}
+        {{- if or (.Values.kubeconfig.enabled) (.Values.volumeMounts) }}
```

Comment:
Probably can be omitted, since volumeMounts  will simply be empty otherwise if not set.

Reply:
The convention [here](https://github.com/prometheus-community/helm-charts/blob/main/charts/kube-state-metrics/templates/deployment.yaml#L131-L134) and elsewhere is not to have an empty key for an array.